### PR TITLE
#1281: Workaround for incorrect equity computation

### PR DESCRIPTION
**Pull request by carlosmiranda**
#1281: See https://github.com/zerocracy/farm/issues/1281#issuecomment-401293773 for explanation of the bug.

The fix involves using the number of minutes, which is always present as a param to `Make payment`. In case `cash` param is absent, the actual cash paid is computed based on the project rate. I patterned this after a similar logic in `make_payment.groovy`: https://github.com/zerocracy/farm/blob/e485549aa173362d9e8f12ccbc9264e929678fa7/src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy#L51-L56

There is a proposed larger fix for inconsistencies in `Make payment` (see #1162). In the meantime this should allow the correct equity to be awarded on future completed tasks.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1294](https://github.com/zerocracy/farm/issues/1294) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1294](https://github.com/zerocracy/farm/issues/1294) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1810653744); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by amihaiemil** [comment]
@carlosmiranda I'll have a look here in 1-2 days, I'm away from my laptop this weekend starting just about now :D

**Comment by g4s8** [comment]
@0crat assign @krzyk

**Comment by g4s8** [comment]
@amihaiemil I hope you don't mind

**Comment by 0crat** [comment]
> @0crat assign @krzyk  ([here](https://github.com/zerocracy/farm/issues/1294#issuecomment-401525538))

@g4s8 This pull request [#1294](https://github.com/zerocracy/farm/issues/1294) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1810813357); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

File: src/main/groovy/com/zerocracy/stk/pm/cost/vesting/transfer_shares.groovy [hunk 1]
```diff
@@ -50,7 +51,12 @@ def exec(Project project, XML xml) {
   if (claim.hasParam('cash')) {
     cash = new Cash.S(claim.param('cash'))
   } else {
-    cash = Cash.ZERO
+    Cash rate = Cash.ZERO
+    Rates rates = new Rates(project).bootstrap()
+    if (rates.exists(login)) {
+      rate = rates.rate(login)
+    }
+    cash = rate.mul(minutes) / 60
```
Comment (krzyk) [inline src/main/groovy/com/zerocracy/stk/pm/cost/vesting/transfer_shares.groovy]: @carlosmiranda There is a consistent stacktrace in appveyor and travis builds related to boosts. When I tried to build myself I also got similar problem (TestListener also failed, but with different error: 1XPath '//paypal/mode/text()' not found in`).

**Comment by carlosmiranda** [comment]
@krzyk please see change, I added code to skip `Payroll` in `make_payment` during testing and added puzzle. This should eliminate Paypal errors.

File: src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy [hunk 2]
```diff
@@ -83,6 +85,17 @@ def exec(Project project, XML xml) {
   Ledger ledger = new Ledger(project).bootstrap()
   String msg
   try {
+    if (new Props(farm).has('//testing')) {
+      // @todo #1281:30min Make payment is currently throws exceptions during
+      //  testing under certain circumstances. It is untestable due to some
+      //  dependency to Paypal payment during testing mode. We should make
+      //  this more testable, perhaps by adding a test method of payments to
+      //  Payroll. That should also make Payroll testable, which it currently
```
Comment (krzyk) [inline src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy]: @carlosmiranda I think this part will be covered by https://github.com/zerocracy/farm/issues/565

**Comment by krzyk** [comment]
@carlosmiranda please see my comments above, it appears I didn't save my github session so I had to readd them.
Travis fails probalby because you have `Matchers.contains(Matchers.containsString(..))` (one of the comments above)

**Comment by carlosmiranda** [comment]
@krzyk you are right. Removed the puzzle and fixed the test (not sure why I had nested `contains` there...)

**Comment by carlosmiranda** [comment]
@krzyk updated, please check

**Comment by krzyk** [comment]
@carlosmiranda xcop doesn't like your formatting of equity.xml, you don't have a new line at the end, and the first set of tags is indented by 4 spaces instead of 2

**Comment by carlosmiranda** [comment]
@krzyk fixed, please check

**Comment by krzyk** [comment]
@carlosmiranda xcop still doesn't like you :( I think the tags should be in single line, but I don't know what's up witht the `<owner id="amihaiemil">0</owner>` tag

**Comment by carlosmiranda** [comment]
@krzyk it's the darned single quotes. Should be fixed now, please check.

**Comment by krzyk** [comment]
@rultor merge

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1294](https://github.com/zerocracy/farm/issues/1294) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +23 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**